src/Symfony/Component/DependencyInjection/Definition.php

```diff
@@ -64,6 +64,13 @@ public function setFactory($factory)
             $factory = explode('::', $factory, 2);
         }
 
+        if (is_array($factory) && !isset($factory[0])) {
```

Comment:
What about explicitly checking for `null`? Why should we support empty arrays here?